Microsoft.Toolkit/Diagnostics/Generated/Guard.Collection.tt

```diff
@@ -0,0 +1,259 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Sounds good! I agree that it might not be immediately clear for everyone what a T4 template even is, or why is all that code being generated, yeah. I'll put together a draft detailing the main points of how the API is structured and I'll put in that `\Generated` folder 👍

Reply:
Done in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3131/commits/caad2f7792c29f83163212b19045a3d3d43d05af 😊